### DISCUSSION TITLE
Custom chain select modal

### DISCUSSION THREAD

**Discussion by chriteixeira** [rainbow-me/rainbowkit#1327] [Q&A] [answered]
We want to use a custom chain modal since we need to show a few other chains not supported by rainbowkit. For these chains (like Solana), we have different adapters to handle them. Because of this, we are not using the chain modal from rainbowkit and we need to programmatically change the chain and disconnected from a wallet.

Is there any way to programmatically switch chains and disconnected from a wallet without using the chain modal and wallet dropdown?

**Comment by Markcode1992** [comment]
Also following

**Comment by DanielSinclair** [comment, marked as answer]
You can use Wagmi's hooks for this: `useDisconnect` and `useSwitchNetwork`. You can also disable the existing RainbowKit chain UI like so:
```tsx
<ConnectButton chainStatus="none" />
```
But Wagmi itself doesn't support Solana, so there may be no direct path to what you're hoping to accomplish.
